account_invoice_pricelist/models/account_invoice.py

```diff
@@ -63,6 +63,13 @@ def _onchange_product_id_account_invoice_pricelist(self):
             product.price, product.taxes_id, self.invoice_line_tax_ids,
             self.company_id)
 
+    @api.onchange('uom_id')
+    def _onchange_uom_id(self):
+        if self.invoice_id.pricelist_id:
```

Comment:
Dont you have to test also self.invoice_id.partner_id ? Like line 49 ?

Reply:
This may not be useful because create invoice line checks that partner exists, and in the later is checked if partner existis to run the same previous onchange.